corehq/apps/app_manager/ui_translations/ui_translations.py

```diff
@@ -46,16 +46,49 @@ def process_ui_translation_upload(app, trans_file):
     default_trans = get_default_translations_for_download(app, commcare_version)
     lang_with_defaults = app.langs[get_index_for_defaults(app.langs)]
 
+    # Find all of the ${0}-looking values in a translation.
+    # Returns list like ["${0}", "${1}", ...]
+    def _get_params(translation):
+        if not translation:
+            return []
+        return re.findall(r"\$.*?}", translation)
+
+    # Create a string version of all parameters in a translation, suitable both for
+    # comparison and for displaying in an error message.
+    def _get_params_text(params):
+        if not params:
+            return "no parameters"
+
+        # Reduce parameter list to the numbers alone, a set like {'0', '1', '2'}
+        numbers = {re.sub(r'\D', '', p) for p in params}
```

Comment:
Yeah, that's okay. Something like `${3rr0r}` will be caught by the "Could not understand" logic below. It will also show up as a mismatch error - like, `Expected ${0}, received {$30}` - but I think that's fine.